### PR TITLE
Add PowerVS OVN Kube routing config

### PR DESCRIPTION
**Pull request by dharaneeshvrd**
**What this PR does / why we need it**:

Fixes https://issues.redhat.com/browse/MULTIARCH-2631
Trying to add routing config for OVN Kube CNI for PowerVS platform 

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go [hunk 1]
```diff
@@ -45,7 +45,15 @@ func ReconcileNetworkOperator(network *operatorv1.Network, networkType hyperv1.N
 				network.Spec.DefaultNetwork.OVNKubernetesConfig.GenevePort = &port
 			}
 		}
+	case hyperv1.PowerVSPlatform:
+		if networkType == hyperv1.OVNKubernetes {
+			if network.Spec.DefaultNetwork.OVNKubernetesConfig == nil {
+				network.Spec.DefaultNetwork.OVNKubernetesConfig = &operatorv1.OVNKubernetesConfig{}
+				network.Spec.DefaultNetwork.OVNKubernetesConfig.GatewayConfig = &operatorv1.GatewayConfig{}
```
Comment (mkumatag) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go:52]:
```suggestion
				network.Spec.DefaultNetwork.OVNKubernetesConfig.GatewayConfig = &operatorv1.GatewayConfig{
				    RoutingViaHost = true
				}
```

Comment (mkumatag) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go:52]: Add some information why we are adding this block, information from this BZ https://bugzilla.redhat.com/show_bug.cgi?id=1996108 can be used here

**Review by mkumatag — CHANGES_REQUESTED** [review]
one minor comment, otherwise lgtm

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go (hunk 1 above)
Comment (dharaneeshvrd) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go]: Done

**Comment by dharaneeshvrd** [comment]
ptal @mkumatag

**Review by mkumatag — APPROVED** [review]
/lgtm

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go [hunk 2]
```diff
@@ -45,7 +45,16 @@ func ReconcileNetworkOperator(network *operatorv1.Network, networkType hyperv1.N
 				network.Spec.DefaultNetwork.OVNKubernetesConfig.GenevePort = &port
 			}
 		}
+	case hyperv1.PowerVSPlatform:
+		if networkType == hyperv1.OVNKubernetes {
+			if network.Spec.DefaultNetwork.OVNKubernetesConfig == nil {
+				network.Spec.DefaultNetwork.OVNKubernetesConfig = &operatorv1.OVNKubernetesConfig{}
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go:51]: why is this needed?

Comment (dharaneeshvrd) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go:51]: Followed the same pattern from KubeVirt platform https://github.com/openshift/hypershift/blob/main/control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go#L41

Initially `network.Spec.DefaultNetwork.OVNKubernetesConfig` is nil and tried to initialise with empty struct and again initialled `GatewayConfig` to set `RoutingViaHost` to true

Comment (dharaneeshvrd) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go:51]: To align with the other part of the block of code refactored slightly.
@enxebre Kept `GatewayConfig` reconciliation inside nil check or else it would try to reassign the same obj on every reconcilation. Please let me know whether that answers your question?

**Comment by dharaneeshvrd** [comment]
@enxebre I think I got your point on reconciling `GatewayConfig` with out the nil check. Moved the reconcilation outside the if. Ptal.

**Review by mkumatag — APPROVED** [review]
/lgtm
